kadi/commands/command_sets.py

```diff
@@ -0,0 +1,263 @@
+            dict(type='COMMAND_HW',
+                 # dur=1.025,
+                 tlmsid='AFIDP',
+                 msid='AFLCRSET'),
+            dict(type='SIMTRANS',
+                 params=dict(POS=-99616),
+                 dur=65.66),
+            dict(type='ACISPKT',
+                 tlmsid='AA00000000',
+                 dur=1.025),
+            dict(type='ACISPKT',
+                 tlmsid='AA00000000',
+                 dur=10.25),
+            dict(type='ACISPKT',
+                 tlmsid=pow_cmd),
+            )
+
+
+def cmd_set_dither(state, date=None):
+    if state not in ('ON', 'OFF'):
+        raise ValueError(f'Invalid dither state {state!r}')
+    enab = 'EN' if state == 'ON' else 'DS'
+    return (dict(type='COMMAND_SW',
+                 tlmsid=f'AO{enab}DITH',
+                 ),
+            )
+
+
+def cmd_set_bright_star_hold(date=None):
+    out = (cmd_set_scs107(date=date)
+           + cmd_set_dither('OFF', date=date)
+           )
+    return out
+
+
+def cmd_set_nsm(date=None):
+    nsm_cmd = dict(type='COMMAND_SW',
+                   tlmsid='AONSMSAF')
+    out = ((nsm_cmd,)
+           + cmd_set_scs107(date=date)
+           + cmd_set_dither('OFF', date=date)
+           )
+    return out
+
+
+def cmd_set_safe_mode(date=None):
+    safe_mode_cmds = (dict(type='COMMAND_SW',
+                           tlmsid='ACPCSFSU'),  # CPE set pcad mode to safe sun
+                      dict(type='COMMAND_SW',
+                           tlmsid='CSELFMT5')  # Format 5 (programmable) select
+                      )
+    # This is a little lazy, but put the NSM commands on top of safe mode.
+    # Even though not 100% accurate this does the right thing for commands and
+    # state processing. Otherwise need to put stuff into commands_v2 and states
+    # to handle the ACPCSFSU command.
+    out = (safe_mode_cmds
+           + cmd_set_nsm(date=date))
+    return out
+
+
+def cmd_set_load_not_run(load_name, date=None):
+    return None
+
+
+def cmd_set_observing_not_run(load_name, date=None):
+    return None
+
+
+def cmd_set_command(*args, date=None):
+    params_str = args[0]
+    cmd_type, args_str = params_str.split('|', 1)
+    cmd = {'type': cmd_type.strip().upper()}
+
+    # Strip spaces around equals signs and uppercase args (note that later the
+    # keys are lowercased).
+    args_str = re.sub(r'\s*=\s*', '=', args_str).upper()
+
+    params = {}
+    for param in args_str.split():
+        key, val = param.split('=')
+        if key == 'TLMSID':
+            cmd['tlmsid'] = val
+        else:
+            params[key] = coerce_type(val)
+    cmd['params'] = params
+
+    return (cmd,)
+
+
+def cmd_set_command_not_run(*args, date=None):
+    cmd, = cmd_set_command(*args, date=date)
+    cmd['type'] = 'NOT_RUN'
+    return (cmd,)
+
+
+def get_cmds_from_event(date, event, params_str):
+    r"""
+    Return a predefined cmd_set ``name`` generated with \*args.
+
+    :param name: cmd set name (manvr|scs107|nsm)
```

Comment:
Not really clear on how these params line up with the args.

Reply:
Good catch, this is a leftover docstring from a previous version, fixed.